react/components/Dropzone/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { PureComponent, createRef } from 'react'
+import PropTypes from 'prop-types'
+import ReactDropZone from 'react-dropzone'
+
+import FileIcon from './FileIcon'
+import Spinner from '../Spinner'
+import ButtonWithIcon from '../ButtonWithIcon'
+import IconClose from '../icon/Close'
+
+const dropzoneRef = createRef()
+
+class Dropzone extends PureComponent {
+  state = {
+    files: [],
+    isHovering: false,
+    fileDropped: false,
+  }
+
+  handleDragEnter = () => {
+    this.setState({ isHovering: true })
+  }
+
+  handleDragLeave = () => {
+    this.setState({ isHovering: false })
+  }
+
+  handleDrop = files => {
+    this.setState({ isHovering: false, fileDropped: true, files })
+    this.props.onDrop(files)
+  }
+
+  handleReset = () => {
+    this.setState({ files: [], fileDropped: false })
+  }
+
+  render() {
+    const { children, isLoading, icon } = this.props
+    const { isHovering, fileDropped, files } = this.state
+    const initialState = !isHovering && !fileDropped
+
+    let dropzoneContainerClasses =
+      'flex flex-column items-center justify-center b--light-blue ba br2 bw1 b--dashed '
+    let iconHolderClasses =
+      'flex items-center justify-center pa6  bg-near-white '
+    const droppedContainerClasses =
+      'bg-light-silver flex flex-row justify-between w-100 items-top'
+
+    const holderSize = '120px'
```

Comment:
Maybe define it as a constant outside of the component, like `HOLDER_SIZE`.

Reply:
Not too sure since it would be the only one used only once. A const file makes sense imo if used in multiple places or several times.

Reply:
Ok, makes sense.